src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -330,49 +331,37 @@ ucs_status_t uct_tcp_sockcm_ep_server_handle_data_received(uct_tcp_sockcm_ep_t *
 {
     uct_tcp_sockcm_priv_data_hdr_t *hdr = (uct_tcp_sockcm_priv_data_hdr_t *)
                                            cep->comm_ctx.buf;
-    uint16_t client_state;
     ucs_status_t status;
     int events;
 
-    if (ucs_test_all_flags(cep->state, UCT_TCP_SOCKCM_EP_SERVER_CONN_CREATED |
-                                       UCT_TCP_SOCKCM_EP_DATA_RECEIVED       |
-                                       UCT_TCP_SOCKCM_EP_CONN_SENT)) {
-        /* the server verifies the client's state */
-        client_state = *(uint16_t*)(hdr + 1);
-        if (client_state != (UCT_TCP_SOCKCM_EP_ON_CLIENT     |
-                             UCT_TCP_SOCKCM_EP_CONN_SENT     |
-                             UCT_TCP_SOCKCM_EP_DATA_RECEIVED |
-                             UCT_TCP_SOCKCM_EP_CLIENT_CONNECTED_CB_INVOKED)) {
-            ucs_error("server received wrong client state = %d", client_state);
-            status = UCS_ERR_IO_ERROR;
-            goto out;
-        }
+    if (cep->state & UCT_TCP_SOCKCM_EP_CONN_SENT) {
```

Comment:
since UCT_TCP_SOCKCM_EP_CONN_SENT defines 2 flags, this condition can be true if data is PACKED, but not SENT yet
is this the intention?

Reply:
why? since it includes both flags, both have to be set for this condition to be true.

Reply:
anyway, I removed UCT_TCP_SOCKCM_EP_CONN_SENT like you suggested in the comment above.

Reply:
actually, `cep->state & (flag_A|flag_B)` is true if wither flag_A or flag_B is set. not only if both are set.